indico/modules/events/registration/schemas.py

```diff
@@ -33,3 +39,86 @@ def sort_list(self, data, many, **kwargs):
         if many:
             data = sorted(data, key=lambda tag: natural_sort_key(tag['title']))
         return data
+
+
+# Schemas for the Check-in app API
+
+class CheckinEventSchema(mm.SQLAlchemyAutoSchema):
+    class Meta:
+        model = Event
+        fields = ('id', 'title', 'description', 'start_dt', 'end_dt')
+
+
+class CheckinRegFormSchema(mm.SQLAlchemyAutoSchema):
+    class Meta:
+        model = RegistrationForm
+        fields = ('id', 'event_id', 'title', 'introduction', 'start_dt', 'end_dt',
+                  'is_open', 'registration_count', 'checked_in_count')
+
+    is_open = fields.Bool(attribute='is_open')
+    registration_count = fields.Int(attribute='existing_registrations_count')
+    checked_in_count = fields.Int(attribute='checked_in_registrations_count')
+
+
+class CheckinRegistrationSchema(mm.SQLAlchemyAutoSchema):
+    class Meta:
+        model = Registration
+        fields = ('id', 'regform_id', 'event_id', 'full_name', 'email', 'state', 'checked_in', 'checked_in_dt',
+                  'checkin_secret', 'is_paid', 'price', 'currency', 'formatted_price', 'tags', 'occupied_slots',
+                  'registration_date', 'registration_data')
+
+    regform_id = fields.Int(attribute='registration_form_id')
+    full_name = fields.Str(attribute='display_full_name')
+    state = fields.Enum(RegistrationState)
+    checkin_secret = fields.UUID(attribute='ticket_uuid')
+    formatted_price = fields.Function(lambda reg: reg.render_price())
+    tags = fields.Function(lambda reg: sorted(t.title for t in reg.tags))
+    registration_date = fields.DateTime(attribute='submitted_dt')
+    registration_data = fields.Method('_get_registration_data')
+
+    def _get_filenames(self, registration):
+        """Extract filenames from file fields."""
+        return {r.field_data.field.id: r.filename
+                for r in registration.data
+                if r.field_data.field.field_impl.is_file_field and r.storage_file_id is not None}
+
+    def _get_registration_data(self, registration):
+        regform = registration.registration_form
+        form_data = get_flat_section_submission_data(regform, registration=registration, management=True)
+        reg_data = get_form_registration_data(regform, registration, management=True)
+        filenames = self._get_filenames(registration)
+        sections = sorted(list(form_data['sections'].values()), key=itemgetter('position'))
+        fields = sorted(list(form_data['items'].values()), key=itemgetter('position'))
```

Comment:
I think the `list()` is not needed since any iterable works

```suggestion
        sections = sorted(form_data['sections'].values(), key=itemgetter('position'))
        fields = sorted(form_data['items'].values(), key=itemgetter('position'))
```